components/plan-build/bin/composite_build_functions.sh

```diff
@@ -152,10 +152,17 @@ _validate_bind_mappings() {
 
     resolved="${resolved_services[$pkg]}"
 
-    while read -r line; do
-      IFS== read bind_name exports <<< "${line}"
-      all_binds_for_pkg[$bind_name]="${exports[@]}"
-    done < <(_read_metadata_file_for "${resolved}" BINDS)
+    if [[ -f "${resolved}/BINDS" ]]; then
+      while read -r line; do
+        IFS== read bind_name exports <<< "${line}"
+        all_binds_for_pkg[$bind_name]="${exports[@]}"
+      done < <(_read_metadata_file_for "${resolved}" BINDS)
+    elif [[ -f "${resolved}/BINDS_OPTIONAL" ]]; then
```

Comment:
I think we want to incorporate _both_ kinds of binds; as coded, if there are `BINDS` _and_ `BINDS_OPTIONAL`, only the former will be read.

Reply:
great point, I'll fix